packages/evm/contracts/Types.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17 <0.9.0;
+
+/**
+ * @title Types - a file that contains all of the type definitions used throughout
+ * the Zodiac Roles Mod.
+ * @author Cristóvão Honorato - <cristovao.honorato@gnosis.pm>
+ * @author Jan-Felix Schwarz  - <jan-felix.schwarz@gnosis.pm>
+ */
+enum ParameterType {
+    None,
+    Static,
+    Dynamic,
+    Tuple,
+    Array,
+    AbiEncoded
+}
+
+enum Operator {
+    // 00:    EMPTY EXPRESSION (default, always passes)
+    //          paramType: Static / Dynamic
+    //          🚫 children
+    //          🚫 compValue
+    /* 00: */ Pass,
+    // ------------------------------------------------------------
+    // 01-04: LOGICAL EXPRESSIONS
+    //          paramType: None
+    //          ✅ children
+    //          🚫 compValue
+    /* 01: */ And,
+    /* 02: */ Or,
+    /* 03: */ Nor,
+    /* 04: */ Xor,
+    // ------------------------------------------------------------
+    // 05-16: COMPLEX EXPRESSIONS
+    //          paramType: AbiEncoded / Tuple / Array,
+    //          ✅ children
+    //          🚫 compValue
+    /* 05: */ Matches,
+    /* 06: */ ArraySome,
+    /* 07: */ ArrayEvery,
+    /* 08: */ ArraySubset,
+    /* 09: */ _ComplexPlaceholder09,
+    /* 10: */ _ComplexPlaceholder10,
+    /* 11: */ _ComplexPlaceholder11,
+    /* 12: */ _ComplexPlaceholder12,
+    /* 13: */ _ComplexPlaceholder13,
+    /* 14: */ _ComplexPlaceholder14,
+    /* 15: */ _ComplexPlaceholder15,
+    /* 16: */ _ComplexPlaceholder16,
+    // ------------------------------------------------------------
+    // 17-31: COMPARISON EXPRESSIONS
+    //          paramType: Static / Dynamic / Tuple / Array / AbiEncoded
+    //          🚫 children
```

Comment:
```suggestion
    //           paramType: Static / Dynamic / Tuple / Array
    //          ❓ children (only for paramType: Tuple / Array to describe their structure)
```

(While technically possible to use EqualTo on AbiEncoded it does not make sense in practice. The param type can be declared as `Dynamic` and the comparison can be performed directly on the bytes instead.)